haystack/nodes/answer_generator/openai.py

```diff
@@ -186,6 +195,23 @@ def __init__(
             logger.debug("Using GPT2TokenizerFast")
             self._hf_tokenizer: PreTrainedTokenizerFast = GPT2TokenizerFast.from_pretrained(tokenizer)
 
+    def using_azure(self) -> bool:
+        return self.deployment_name is not None and self.base_url is not None
+
+    def resolve_url(self) -> str:
```

Comment:
Same here I think we could make this method private by adding the `_` to the beginning of the name.